src/Assertions.jl

```diff
@@ -89,6 +86,64 @@ function _global_indent()
   return s
 end
 
+@doc Markdown.doc"""
+    @vprint(S::Symbol, k::Int, msg::String)
+    @vprint S k msg
+
+    @vprint(S::Symbol, msg::String)
+    @vprint S msg
+
+This macro can be used to control printings inside the code.
+
+The macro `vprint` has two or three entry arguments: a symbol `S` refered as a
+*verbose scope*, an optional integer `k` and a string `msg`. If `k` is not
```

Comment:
This is of course not your fault, but: "verbose scope" and "verbose level" IMHO really should be "verbosity scope" and "verbosity level". But of course that's something @thofma and @fieker should decide.